src/dotnet-aspnet-codegenerator/Program.cs

```diff
@@ -6,127 +6,242 @@
+
+
                 var configuration = appConfiguration.Value() ?? "Debug";
-                if(configuration != null && !configuration.Equals("Release") && !configuration.Equals("Debug")) 
+                // If its not a dependency command, use netcoreapp1.0 (ignore what is passed in). 
+                NuGetFramework nugetFramework = FrameworkConstants.CommonFrameworks.NetCoreApp10;
+
+                if (isProjectDependencyCommand && !TryGetNugetFramework(framework.Value(), out nugetFramework))
+                {
+                    throw new ArgumentException($"Could not understand the NuGetFramework information. Framework short folder name passed in was: {framework.Value()}");
+                }
+
+                var projectContextInfo = GetProjectContextInformation(projectPath.Value(), nugetFramework);
+
+                if (isProjectDependencyCommand && projectContextInfo.ProjectContext == null)
                 {
-                    throw new ArgumentException($"Invalid value for configuration: {configuration}. {appConfiguration.Description}");
+                    // We tried to invoke dependency command with the framework from the project.json and still failed. 
+                    var msg = $"Could not get project context for project with framework {nugetFramework.GetShortFolderName()}";
+                    _logger.LogMessage(msg, LogMessageLevel.Error);
+                    throw new Exception(msg);
                 }
-                Directory.SetCurrentDirectory(context.ProjectDirectory);
-                AddFrameworkServices(serviceProvider, context, packagesPath.Value());
-                AddCodeGenerationServices(serviceProvider);
-                var codeGenCommand = serviceProvider.GetService<CodeGenCommand>();
-                codeGenCommand.Execute(app.RemainingArguments.ToArray());
-                return 0;
+                else if (projectContextInfo.ProjectContext == null)
+                {
+
+                    var frameworkToUse = projectContextInfo.TargetFrameworks.FirstOrDefault();
+                    // We tried to get the project context for netcoreapp1.0 and failed. 
+                    // We need to invoke project Dependency command.
+                    var buildExitCode = DotNetBuildCommandHelper.Build(
+                        projectPath.Value(),
+                        configuration,
+                        frameworkToUse,
+                        buildBasePath.Value(),
+                        outputPath.Value());
+
+                    if (buildExitCode != 0)
+                    {
+                        //Build failed. 
+                        // Stop the process here. 
+                        return buildExitCode;
+                    }
+
+                    // Invoke the dependency command
+                    var projectDirectory = Directory.GetCurrentDirectory();
```

Comment:
Curious then, why do we ever need to expect `ProjectContext` to be null? `ProjectContext` construction only relies on a project being restored (a lock file). And that's the same expectation as the rest of the code here. 

Just asking because always creating a `ProjectContext` could simplify a lot of what you have here. Unless I'm missing something of course :smile: